xarray/groupers.py

```diff
@@ -380,6 +380,13 @@ def _init_properties(self, group: T_Group) -> None:
         if isinstance(group_as_index, CFTimeIndex):
             from xarray.core.resample_cftime import CFTimeGrouper
 
+            if not isinstance(self.freq, str | BaseCFTimeOffset):
```

Comment:
It seems possible to cast instead of raising an error here. But that could be addressed in a future PR.

Reply:
I've had a crack at doing this, just need to throw together some tests for it. Let me know if there's anything you would like changed.